src/history/StateSnapshot.cpp

```diff
@@ -109,22 +98,6 @@ StateSnapshot::writeHistoryBlocks() const
     // When writing checkpoint 0x3f (63) we will have written 63 headers because
     // header 0 doesn't exist, ledger 1 is the first. For all later checkpoints
     // we will write 64 headers; any less and something went wrong[1].
-    //
-    // [1]: Probably our read transaction was serialized ahead of the write
-    // transaction composing the history itself, despite occurring in the
-    // opposite wall-clock order, this is legal behavior in SERIALIZABLE
-    // transaction-isolation level -- the highest offered! -- as txns only have
-    // to be applied in isolation and in _some_ order, not the wall-clock order
-    // we issued them. Anyway this is transient and should go away upon retry.
```

Comment:
Tbh, I'm not certain what the guarantees are exactly wrt SCP messages - my understanding is that we can be missing arbitrary ledgers (if we got in sync via catchup, for example). I agree it might be nice to strengthen this check, but I think it's probably out of scope for this PR, since I'm just preserving the behavior in master for now.

Reply:
yeah, fair enough. I just noticed and thought it might be worth keeping some kind of comment around explaining that this is possible. but I guess it's not that important.